File: src/net/eduard/api/server/BungeeDB.java
package net.eduard.api.server;

import java.util.UUID;

import net.eduard.api.lib.manager.DBManager;

public class BungeeDB {

	private DBManager manager;

	public BungeeDB(DBManager manager) {
		setManager(manager);
	}
	public void createBungeeTables() {
		getDB().createTable("servers",
				"name varchar(50), host varchar(100), port int, players int, status int");
		getDB().createTable("players", "name varchar(16), uuid varchar(100), server varchar(50)");
	}

	/**
	 * Adiciona um numero atual do tempo na tabela para o jogador
	 * 
	 * @param table
	 *            Tabela
	 * @param player
	 *            Jogador
	 */
	public void addCooldown(String table, String playerId) {
		getDB().createTable(table, "uuid varchar not null, time long not null");

		if (!getDB().contains(table, "uuid = ?", playerId)) {
			getDB().insert(table, playerId, System.currentTimeMillis());
		} else {
			getDB().change(table, "id = ?", "time = ?", playerId, System.currentTimeMillis());
		}
	}

	public void setStatusServer(String server, int status) {
		getDB().change("servers", "status = ?", "name = ?", status, server);
	}
	public void setPlayerServer(UUID playerId,String serverName) {
		getDB().change("players", "server = ?", "uuid = ?", serverName,
				playerId);
	}

	public boolean isOnServer(String server, String playerName) {
		return getDB().contains("players", "name = ? and server = ?", playerName, server);
	}

	public String getPlayerServer(UUID playerId) {
		return getDB().getString("players", "server", "uuid = ?", playerId);
	}

	public boolean playersContains(String name) {
		return getDB().contains("players", "name = ?", name);
	}

	public boolean playersContains(UUID playerId) {
		return getDB().contains("players", "uuid = ?", playerId);
	}

	public boolean serversContains(String name) {
		return getDB().contains("servers", "name = ?", name);
	}

	public void setPlayersAmount(String server, int amount) {
		getDB().change("servers", "players = ?", "name = ?", amount, server);
	}

	public int getPlayersAmount(String server) {
		return getDB().getInt("servers", "players", "name = ?", server);
	}

	public DBManager getDB() {
		return getManager();
	}

	public DBManager getManager() {
		return manager;
	}

	public void setManager(DBManager manager) {
		this.manager = manager;
	}

}

File: src/net/eduard/api/server/BungeeDB.kt
package net.eduard.api.server

import java.util.UUID

import net.eduard.api.lib.database.DBManager

@SuppressWarnings("unused")
class BungeeDB(var db: DBManager) {

    val serverTable = "network_servers"
    val playerTable = "network_players"

    fun createNetworkTables() {
        db.createTable(serverTable,
                "name varchar(50), host varchar(100), port int, players int, status int")
        db.createTable(playerTable, "name varchar(16), uuid varchar(100), server varchar(50)")
    }

    fun setStatusServer(server: String, status: Int) {
        db.change(serverTable, "status = ?", "name = ?", status, server)
    }

    fun setPlayerServer(playerId: UUID, serverName: String) {
        db.change(serverTable, "server = ?", "uuid = ?", serverName,
                playerId)
    }

    fun isOnServer(server: String, playerName: String): Boolean {
        return db.contains(serverTable, "name = ? and server = ?", playerName, server)
    }

    fun getPlayerServer(playerId: UUID): String {

        val data = db.select("select * from $playerTable where uuid = ?",playerId)
        var server = "lobby"
        if (data.next()){
            server = data.getString("server")
        }
        data.close();

        return  server;
    }

    fun playersContains(name: String): Boolean {
        return db.contains(playerTable, "name = ?", name)
    }

    fun playersContains(playerId: UUID): Boolean {
        return db.contains(playerTable, "uuid = ?", playerId)
    }

    fun serversContains(name: String): Boolean {
        return db.contains(serverTable, "name = ?", name)
    }

    fun setPlayersAmount(server: String, amount: Int) {
        db.change(serverTable, "players = ?", "name = ?", amount, server)
    }

    fun getPlayersAmount(server: String): Int {

        val data = db.select("select * from $serverTable where name = ?",server)
        var amount = 0
        if (data.next()){
            amount = data.getInt("players")
        }
        data.close();

        return  amount;
    }

}
